core/src/main/java/org/apache/iceberg/util/JsonUtil.java

```diff
@@ -184,6 +184,11 @@ public static String getStringOrNull(String property, JsonNode node) {
     return pNode.asText();
   }
 
+  public static JsonNode get(String property, JsonNode node) {
+    Preconditions.checkArgument(node.has(property), "Cannot parse missing object %s", property);
```

Comment:
How do you know that the missing property needed to be an object? I think this should be `Cannot parse, missing field: %s`